executioncontext/projectview/src/test/java/org/jetbrains/bsp/bazel/projectview/parser/sections/ProjectViewSingletonSectionParserTest.java

```diff
@@ -22,191 +23,205 @@
-    var sectionName = parser.getSectionName();
-
-    return Arguments.of(parser, rawValueConstructor, sectionConstructor, sectionName);
-  }
-
-  private static Arguments debuggerAddressSectionArguments() {
-    var parser = ProjectViewDebuggerAddressSectionParser.INSTANCE;
-    var rawValueConstructor = (Function<String, String>) (seed) -> "host_" + seed + ":8080";
-    var sectionMapper =
-        (Function<String, ProjectViewDebuggerAddressSection>)
-            ProjectViewDebuggerAddressSection::new;
-    var elementMapper = (Function<String, String>) x -> x;
-
-    var sectionConstructor =
-        createSectionConstructor(rawValueConstructor, sectionMapper, elementMapper);
-    var sectionName = parser.getSectionName();
-
-    return Arguments.of(parser, rawValueConstructor, sectionConstructor, sectionName);
-  }
-
-  private static Arguments javaPathSectionArguments() {
-    var parser = ProjectViewJavaPathSectionParser.INSTANCE;
-    var rawValueConstructor = (Function<String, String>) (seed) -> "/path/to/java/" + seed;
-    var sectionMapper =
-        (Function<Path, ProjectViewJavaPathSection>) ProjectViewJavaPathSection::new;
-    var elementMapper = (Function<String, Path>) Paths::get;
-
-    var sectionConstructor =
-        createSectionConstructor(rawValueConstructor, sectionMapper, elementMapper);
-    var sectionName = parser.getSectionName();
-
-    return Arguments.of(parser, rawValueConstructor, sectionConstructor, sectionName);
-  }
-
-  private static <V, T extends ProjectViewSingletonSection<V>>
-      Function<String, T> createSectionConstructor(
-          Function<String, String> rawValueConstructor,
-          Function<V, T> sectionMapper,
-          Function<String, V> elementMapper) {
-
-    return (seed) -> sectionMapper.apply(elementMapper.apply(rawValueConstructor.apply(seed)));
-  }
-
-  @Nested
-  @DisplayName("T parse(rawSection) tests")
-  class ParseRawSectionTest {
-
-    @MethodSource(
-        "org.jetbrains.bsp.bazel.projectview.parser.sections.ProjectViewSingletonSectionParserTest#data")
-    @ParameterizedTest
-    public void shouldReturnFailureForWrongSectionName(
-        ProjectViewSingletonSectionParser<V, T> parser,
-        Function<String, String> rawValueConstructor,
-        Function<String, T> sectionConstructor,
-        String sectionName) {
-      // given
-      var rawSection = new ProjectViewRawSection("wrongsection", "value");
-
-      // when
-      var sectionTry = parser.parse(rawSection);
-
-      // then
-      assertThat(sectionTry.isFailure()).isTrue();
-      assertThat(sectionTry.getCause().getClass()).isEqualTo(IllegalArgumentException.class);
-      assertThat(sectionTry.getCause().getMessage())
-          .isEqualTo(
-              "Project view parsing failed! Expected '"
-                  + sectionName
-                  + "' section name, got 'wrongsection'!");
+    public static Stream<Arguments> data() {
```

Comment:
something with formatting happened in this file as well,

Reply:
I saw that this file is wanted to be switched to Kotlin, I can do that if you wish

Reply:
lets keep it as java for now